ui/src/shared/components/TagInput.tsx

```diff
@@ -1,34 +1,25 @@
-import React, {Component} from 'react'
-import PropTypes from 'prop-types'
+import React, {PureComponent} from 'react'
 import _ from 'lodash'
 
-import Tags from 'shared/components/Tags'
+import Tags from 'src/shared/components/Tags'
+import {Input} from 'src/types/kapacitor'
 
-class TagInput extends Component {
-  handleAddTag = e => {
-    if (e.key === 'Enter') {
-      e.preventDefault()
-      const newItem = e.target.value.trim()
-      const {tags, onAddTag} = this.props
-      if (!this.shouldAddToList(newItem, tags)) {
-        return
-      }
-
-      this.input.value = ''
-      onAddTag(newItem)
-      this.props.disableTest()
-    }
-  }
-
-  handleDeleteTag = item => {
-    this.props.onDeleteTag(item)
-  }
+interface Item {
```

Comment:
seems like it would be better to choose either text or name and enforce that usage everywhere, without it being conditional, if possible